functions-iterators/starter.js

```diff
@@ -29,53 +29,213 @@ This will help us track your progress and it will give you practice writing good
 
 // PART 0: Just DO it!!
 
-
+function justDoIt(string){
+  return `${string}, JUST DO IT`;
+}
+//console.log(justDoIt('Abdulrab'));
 
 // PART 1: Big or Small String?
 
-
+function bigOrSmallString(string){
+  if (string.length > 10){
+    return `This word is loooooong!`;
+  } else {
+    return `This word is short 😬`;
+  }
+}
+//console.log(bigOrSmallString("Abdulrab"));
 
 // PART 2: Odd or Even String Length?
 
-
+function oddOrEvenString(string){
+  if (string.length % 2 == 0){
+    return `This string length is TOTALLY even! : ${string.length}`;
+  } else {
+    return `This string's length is odd. : ${string.length}`;
+  }
+}
+//console.log(oddOrEvenString('Abdulrab'));
 
 // PART 3: Median
-
-
+var number1 = [89, 20, 40, 30, 50, 77, 10];
+
+number1.sort(function(a, b){return a-b});
+
+function medianOcfArray(arr){
+  var median;
+  if (arr.length % 2 == 1){
+    median = (arr.length - 1) / 2;
+    return arr[median];
+  } else {
+    console.log('The lenght of array is event.')
+  }
```

Comment:
this is not the way to do median you need to work on your logic your code should look like this 

```
 function medianOcfArray(arr){
  var median;
  if ((arr.length % 2) == 0){
    median = Math.floor(arr.length / 2);

    return ((arr[median] + arr[median - 1]) / 2);
  } else {
    //  console.log('The lenght of array is event.')
    median = Math.floor(arr.length / 2)
    return arr[median]
  }
}
```